### DISCUSSION TITLE
"代码执行"  节点导入第三方包受限

### DISCUSSION THREAD

**Discussion by hnliuhong** [langgenius/dify#3486] [Help]
代码执行节点非常好，它可以让节点更匹配我们的业务。测试时发现如果导入一些第三方模块例如python中导入：from urllib3 import HTTPResponse 则会提示 run failed: error: operation not permitted。从提示来看是操作受限！请问如何解决或是否有替代性解决方案。很多时候导入第三方包是必须的

**Comment by songdo** [comment]
有答案了吗

**Comment by sharptornadoqsh** [comment]
看sandbox的文档，是因为一些包的执行需要调用linux的系统函数，sandbox本身写入了一些系统调用函数，这些函数支持基本上python自定义的包。如果添加新包，则要查看新包需要哪些系统调用，添加到sandbox的一些文件中，重新buildsandbox镜像，可以看下sandbox的faq

**Reply by shixiang1011** [reply]
您好，请问具体操作方法有流程吗

**Reply by sharptornadoqsh** [reply]
https://github.com/langgenius/dify-sandbox/blob/main/FAQ.md
参考这个，修改源码，重新打镜像，把所有的linux系统权限都打开，或者查看你添加的库需要哪些linux权限，都打开再编译

**Comment by Tommy-Mo** [comment]
有答案了吗

**Comment by sharptornadoqsh** [comment]
您好，您的邮件我已经收到，我会尽快处理

**Comment by laisx** [comment]
nodejs也是不能引入任何的包。相比于coze这个功能有点鸡肋了。coze就能安装包

**Comment by sharptornadoqsh** [comment]
您好，您的邮件我已经收到，我会尽快处理

**Comment by sharptornadoqsh** [comment]
您好，您的邮件我已经收到，我会尽快处理
